rust/connlib/snownet/src/node.rs

```diff
@@ -123,7 +125,12 @@ where
     TId: Eq + Hash + Copy + Ord + fmt::Display,
     RId: Copy + Eq + Hash + PartialEq + Ord + fmt::Debug + fmt::Display,
 {
-    pub fn new(private_key: StaticSecret, buf_size: usize, seed: [u8; 32]) -> Self {
+    pub fn new(
+        private_key: StaticSecret,
+        buf_size: usize,
+        seed: [u8; 32],
+        max_idle: Duration,
```

Comment:
What about:

- Make `idle_connection_timeout` an `Option`
- Default to `None`
- Add a `with_idle_connection_timeout` function to `ClientNode`

That should reduce the surface area of this change a fair bit and we can still easily configure it as part of `ClientState`.